### PR TITLE
Implement a method that sorts an integer slice using BubbleSort

### PR DESCRIPTION
**Pull request by mohankrishna**
Fixes #42 

#### Changes proposed in this PR:

- Add 02_bubble/main.go for implementing BubbleSort to an integer slice
- Add 02_bubble/main_test.go which contains unit tests to verify BubbleSort implementation

### CODE REVIEW

**Comment by paraganz** [comment]
Test code coverage is not 100%. Hence, build is failing. Please check.

File: 02_bubble/mohankrishna/main.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) (sorted []int) {
+	sorted = make([]int, len(s))
+	copy(sorted, s)
+	swapped := true
+	for swapped {
```
Comment (anzdaddy) [inline 02_bubble/mohankrishna/main.go]: Initialise `swapped` in the for-loop.
```go
for swapped := true; swapped; {
```

File: 02_bubble/mohankrishna/main.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		swapped = false
+		for i := 0; i < len(sorted)-1; i++ {
+			if sorted[i+1] < sorted[i] {
+				swapped = true
+				sorted[i+1], sorted[i] = sorted[i], sorted[i+1]
```
Comment (anzdaddy) [inline 02_bubble/mohankrishna/main.go:24]: 👍

File: 02_bubble/mohankrishna/main.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) (sorted []int) {
```
Comment (anzdaddy) [inline 02_bubble/mohankrishna/main.go]: Avoid named return values unless the situation demands it or it aids documentation.

File: 02_bubble/mohankrishna/main.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+			if sorted[i+1] < sorted[i] {
+				swapped = true
+				sorted[i+1], sorted[i] = sorted[i], sorted[i+1]
+			}
+		}
+	}
+	return
+}
+
+func insertionSort(s []int) (sorted []int) {
```
Comment (anzdaddy) [inline 02_bubble/mohankrishna/main.go]: Avoid named return values unless the situation demands it or it aids documentation.

File: 02_bubble/mohankrishna/main.go (hunk 3 above)
Comment (paraganz) [inline 02_bubble/mohankrishna/main.go]: Can you please change the input array name as it looks like array is alread sorted?

File: 02_bubble/mohankrishna/main.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	for swapped {
+		swapped = false
+		for i := 0; i < len(sorted)-1; i++ {
+			if sorted[i+1] < sorted[i] {
+				swapped = true
+				sorted[i+1], sorted[i] = sorted[i], sorted[i+1]
+			}
+		}
+	}
+	return
```
Comment (paraganz) [inline 02_bubble/mohankrishna/main.go]: Do we need return here as this is last line?

File: 02_bubble/mohankrishna/main.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	sorted = make([]int, len(s))
+	copy(sorted, s)
+	for i := 1; i < len(sorted); i++ {
+		for j := i - 1; j >= 0; j-- {
+			if sorted[j] > sorted[j+1] {
+				sorted[j+1], sorted[j] = sorted[j], sorted[j+1]
+			}
+		}
+	}
+	return
```
Comment (paraganz) [inline 02_bubble/mohankrishna/main.go]: Same as above

File: 02_bubble/mohankrishna/main.go (hunk 5 above)
Comment (mohankrishna) [reply 02_bubble/mohankrishna/main.go]: Yes, it is mandatory.

File: 02_bubble/mohankrishna/main.go (hunk 6 above)
Comment (mohankrishna) [reply 02_bubble/mohankrishna/main.go]: Yes, it is mandatory.

**Review by anzdaddy — CHANGES_REQUESTED** [review]

File: 02_bubble/mohankrishna/main.go (hunk 1 above)
Comment (mohankrishna) [reply 02_bubble/mohankrishna/main.go]: Updated!

File: 02_bubble/mohankrishna/main.go (hunk 3 above)
Comment (mohankrishna) [reply 02_bubble/mohankrishna/main.go]: Done!

File: 02_bubble/mohankrishna/main.go (hunk 4 above)
Comment (mohankrishna) [reply 02_bubble/mohankrishna/main.go]: Done!

**Review by juliaogris — APPROVED** [review]
LGTM - just gotta get @anzdaddy to agree as well. Ping him on Slack - he never sleeps ;)

**Review by anzdaddy — APPROVED** [review]

**Pull request merged**